src/routes/PaymentResponsePageV2.tsx

```diff
@@ -0,0 +1,146 @@
+import PageContainer from "../components/PageContent/PageContainer";
+import { responseOutcome } from "../features/payment/models/responseOutcome";
+import { useAppDispatch } from "../redux/hooks/hooks";
+import { onBrowserUnload } from "../utils/eventListeners";
+import { moneyFormat } from "../utils/form/formatters";
+import {
+  clearStorage,
+  getSessionItem,
+  SessionItems,
+} from "../utils/storage/sessionStorage";
+import { ViewOutcomeEnum } from "../utils/transactions/TransactionResultUtil";
+import { Cart } from "../features/payment/models/paymentModel";
+import { NewTransactionResponse } from "../../generated/definitions/payment-ecommerce/NewTransactionResponse";
+import { resetThreshold } from "../redux/slices/threshold";
+import { Bundle } from "../../generated/definitions/payment-ecommerce/Bundle";
+import { ROUTE_FRAGMENT } from "./models/routeModel";
+
+type PrintData = {
+  useremail: string;
+  amount: string;
+};
+
+export default function PaymentResponsePageV2() {
+  const outcome = getFragmentParameter(
+    window.location.href,
+    ROUTE_FRAGMENT.OUTCOME,
+    false
+  ) as ViewOutcomeEnum;
+
+  const conf = getConfigOrThrow();
+
+  const outcomeMessage = responseOutcome[outcome];
+
+  const cart = getSessionItem(SessionItems.cart) as Cart | undefined;
+
+  const redirectUrl =
+    outcome === "0"
+      ? cart?.returnUrls.returnOkUrl || "/"
+      : cart?.returnUrls.returnErrorUrl || "/";
+
+  const transactionData = getSessionItem(SessionItems.transaction) as
+    | NewTransactionResponse
+    | undefined;
+
+  const pspSelected = getSessionItem(SessionItems.pspSelected) as
+    | Bundle
+    | undefined;
+
+  const email = getSessionItem(SessionItems.useremail) as string | undefined;
+
+  const totalAmount =
+    Number(
+      transactionData?.payments
+        .map((p) => p.amount)
+        // eslint-disable-next-line @typescript-eslint/restrict-plus-operands
+        .reduce((sum, current) => sum + current, 0)
+    ) + Number(pspSelected?.taxPayerFee);
+
+  const usefulPrintData: PrintData = {
+    useremail: email || "",
+    amount: moneyFormat(totalAmount),
+  };
+
+  const dispatch = useAppDispatch();
+
+  useEffect(() => {
+    dispatch(resetThreshold());
+    window.removeEventListener("beforeunload", onBrowserUnload);
+    clearStorage();
+  }, []);
+
+  const { t } = useTranslation();
+
+  return (
+    <PageContainer>
+      <Box
+        sx={{
+          display: "flex",
+          alignItems: "center",
+          py: 5,
+        }}
+      >
+        <Box
+          sx={{
+            py: 5,
+            display: "flex",
+            justifyContent: "center",
+            width: "100%",
+            flexDirection: "column",
+            alignItems: "center",
+          }}
+        >
+          <img
+            src={outcomeMessage?.icon}
+            alt="cancelled"
+            style={{ width: "80px", height: "80px" }}
+          />
+          <Typography
```

Comment:
You could use a separate component or function to avoid repeating the logic that renders the usefulPrintData

Reply:
In this scenario the effort to create an extra component is not justified. Thanks for the suggestion anyway